components/prism-javascript.js

```diff
@@ -76,6 +76,31 @@ Prism.languages.insertBefore('javascript', 'string', {
 	}
 });
 
+Prism.languages.insertBefore("javascript", "template-string", {
+  "styled-components-template-string": {
+    pattern: /(styled(\.\w+|\([^\)]*\))(\.\w+(\([^\)]*\))*)*|css|createGlobalStyle|keyframes)`(?:\$\{[^}]+\}|\\\\|\\?[^\\])*?`/,
```

Comment:
1. Please don't use capturing groups if not necessary.
2. You don't need to escape `)` in a char set.

A corrected minimal pattern could look like this: `styled(?:\([^)]*\))?(?:\.\w+(?:\([^)]*\))*)+` (plus the rest)

Reply:
That specific RegExp does not correctly match

```JS
styled(Button)`
  background: red;
`
```

Results in this test failure:

```
  1) Testing language 'javascript' – should pass test case 'styled-components_feature':

      + expected - actual

           "punctuation"
           ")"
         ]
         [
      -    "template-string"
      +    "styled-components-template-string"
           [
             [
               "string"
      -        "`\n  background: red;\n`"
      +        "`\n  background: red"
             ]
      +      ";"
      +      [
      +        "string"
      +        "\n`"
      +      ]
           ]
         ]
```

Reply:
You're right, my bad. The last `+` has to be a `*`, so `styled(?:\([^)]*\))?(?:\.\w+(?:\([^)]*\))*)*`.